pkg/transaction/sender_matcher.go

```diff
@@ -106,6 +106,12 @@ func (m *Matcher) Matches(ctx context.Context, tx []byte, networkID uint64, send
 		return nil, fmt.Errorf("%v: %w", err, ErrTransactionSenderInvalid)
 	}
 
+	var type2sender common.Address
+
+	if len(nTx.Data()) == 20 {
```

Comment:
can we have a small comment about how this works? can't really understand when this data is supposed to be `nil` or not, and the variable names could also be improved. what does type 1 mean and what does type 2 mean?

Reply:
logic changed a bit. names should now be clearer and comments were added.